mongodbatlas/fw_resource_mongodbatlas_search_deployment.go

```diff
@@ -366,3 +384,21 @@ func newTFSpecsModel(specs []admin.ApiSearchDeploymentSpec) []tfSearchNodeSpecMo
 
 	return result
 }
+
+func NewSearchDeploymentServiceFromClient(client *admin.APIClient) SearchDeploymentService {
+	return &SearchDeploymentServiceFromClient{
+		client: client,
+	}
+}
+
+type SearchDeploymentService interface {
+	GetAtlasSearchDeployment(ctx context.Context, groupID, clusterName string) (*admin.ApiSearchDeploymentResponse, *http.Response, error)
+}
+
+type SearchDeploymentServiceFromClient struct {
+	client *admin.APIClient
+}
+
+func (a *SearchDeploymentServiceFromClient) GetAtlasSearchDeployment(ctx context.Context, groupID, clusterName string) (*admin.ApiSearchDeploymentResponse, *http.Response, error) {
+	return a.client.AtlasSearchApi.GetAtlasSearchDeployment(ctx, groupID, clusterName).Execute()
+}
```

Comment:
Was unit testing the only reason why you define the interface here?

Reply:
Yes. If we define a separate package structure can see this defined in a separate file to avoid noise in the main implementation file. Did you have any particular concern?

Reply:
no, I was wondering if I was missing other reasons. I agree that a separate package structure would help here